### ISSUE TITLE
MixedLM summary: z or t?

### ISSUE THREAD

**Issue by uliano** [statsmodels/statsmodels#2620]
In this page:

http://statsmodels.sourceforge.net/stable/examples/notebooks/generated/mixed_lm_example.html

statsmodels and R are compared side by side. however in the summary for fixed effects what R calls t value in statsmodels is labeled as z; accordingly the P>|z| should probably be P>|t|

Is there a reason for that?

**Comment by josef-pkt** [comment]
MixedLM uses the normal distribution and not the t distribution, so the `p>|z|` is correct.
In this case it is hardcoded and there is no option to change to `use_t=True`. (I guess the keyword for `t_test` could be used to generate a params table with t based p-values.)

I never looked at the details of whether t would be a good approximation to the small sample distribution. It's @kshedden's decision.
For example I saw recently this link in a tweet
https://stat.ethz.ch/pipermail/r-help/2006-May/094765.html


In general, in almost all models we only have theoretical results based on asymptotic normality. In some cases t (and F) distribution provide a better small sample approximation.
So we default to using the normal distribution (everywhere except for the RegressionModels OLS, ...). But the generic setup now allows the user to choose the distribution with `use_t`. However, this does not work for models like MixedLM that partially overwrite the generic inference methods.

I think we could or should add the use_t option also to MixedLM.

**Comment by kshedden** [comment]
I was talking to someone about this the other day.  Once we get into providing t probabilities we have to deal with the degrees of freedom.  There are many approaches, n-p, Satterthwaite, Kenward-Roger, etc.  It should be added at some point but it involves some work.  

Right now the reported probabilities are based on the normal approximation so it is appropriate to use P > |z|.

**Comment by uliano** [comment]
So the fact that I get the same exact value for z in MixedLM as the value for t in lmer is due to the high number of degree of freedom that makes t the same as z?

**Comment by josef-pkt** [comment]
I just saw http://stats.stackexchange.com/questions/118416/getting-p-value-with-mixed-effect-with-lme4-package
where one answer list a few of these options for small sample approximation to the distribution.

**Comment by josef-pkt** [comment]
> So the fact that I get the same exact value for z in MixedLM as the value for t in lmer is due to the high number of degree of freedom that makes t the same as z?

Yes, with large degrees of freedom there is not much of a difference except for maybe some values far in the tail, asymptotically they are the same.

However, depending on the structure of the data the overall sample size and degrees of freedom might not provide the correct small sample approximation.
I don't know about the MixedLM case, but for cluster robust standard errors one important requirement for the data is to have many clusters. Our (and Stata's) small sample correction uses the degrees of freedom for the small sample t/F approximation depending on the number of clusters and not depending on the total sample size.
This is for example relevant if a parameter is mainly identified by cross-cluster variation and not within cluster variation, and we have many observations per cluster but only a few clusters.

**Comment by kshedden** [comment]
@uliano what is your sample size, how many fixed effects df, and any information about the random effects structure?  Exact agreement is not expected but I actually don't know exactly what R does.

If your sample size is in the hundreds with a relatively simple model it could appear exact as printed in the summary.

**Comment by uliano** [comment]
@kshedden I haven't my sample, yet. I was walkin' down the example in the page linked here at the top of discussion, however on R I linked the lmerTest Library that gives t values and p>|t|. The values are identical up to all printed digit. And yes, sample size is BIG!

**Comment by josef-pkt** [comment]
The first topic here has the view of the R developers (and is the source for several answers on stats.stackexchange) http://glmm.wikidot.com/faq

**Comment by uliano** [comment]
@josef-pkt I'm aware about the lack of consensus on p values for mixed effect models, however outside in the "real world" they still want some p to publish your paper and don't seem to be concerned too much about statistician's subtleties rather it is enough to cite some commercial black box program that gave you the damned p. If SAS mixed proc uses Satterthwaite for me that is "validated" so I use lmerTest that does the same. I'm somewhere between hammer (head of the lab) and anvil (reviewers), at the moment I'm struggling with R and as I'm quite fluent in Python I am considering a switch to pandas/statsmodels.

**Comment by josef-pkt** [comment]
@uliano  I wasn't aware of the pvalue issue and possible options for the Mixed models cases. (So, FAQ and links are also for my own benefit.)

About switching to python: Depending on the methods that you use, it will not be possible to switch  completely to python. There are still many methods and options not available without coding them yourself, so R/rpy or SAS or Stata or ... are still necessary to complement the python packages.

"If the other packages do it this way, then we can also use it", is most of the time a useful approach which makes also writing unit tests easier.
But it's still work that needs to be done.

SAS description for implementing the degrees of freedom in MIXED and GLIMMIX
http://support.sas.com/documentation/cdl/en/statug/63033/HTML/default/viewer.htm#statug_mixed_sect015.htm#statug.mixed.mixedmodelddfm
http://support.sas.com/documentation/cdl/en/statug/63962/HTML/default/viewer.htm#statug_glimmix_a0000001441.htm
(looks easy enough, if we had `g`, `C`, and `A` already available, but not otherwise.)
